### PR TITLE
Fixing makerdao subgraph

### PR DESCRIPTION
**Pull request by tnkrxyz**
This PR fixes makerdao subgraph issues listed in #500, improve code readability, organization and documentation.
- [x] Debug deployment failure
- [x] Verify issues fixed (commented in #500)

### CODE REVIEW

File: subgraphs/makerdao/src/utils/numbers.ts [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+  BIGINT_ZERO,
+  BIGDECIMAL_TWO,
+  BIGDECIMAL_ONE,
+  BIGDECIMAL_SIX,
+  BIGDECIMAL_TWELVE,
+  BIGDECIMAL_THREE,
+  BIGDECIMAL_ONE_THOUSAND,
+} from "../common/constants";
+
+export function bigIntToBDUseDecimals(quantity: BigInt, decimals: i32 = 18): BigDecimal {
```
Comment (this-username-is-taken) [inline subgraphs/makerdao/src/utils/numbers.ts:15]: This is nice. Can you add this to the common lib in the reference subgraph?

File: subgraphs/makerdao/src/utils/numbers.ts [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+  return quantity.divDecimal(
+    BigInt.fromI32(10)
+      .pow(decimals as u8)
+      .toBigDecimal(),
+  );
+}
+
+export function bigDecimalExponential(rate: BigDecimal, exponent: BigDecimal): BigDecimal {
```
Comment (this-username-is-taken) [inline subgraphs/makerdao/src/utils/numbers.ts:23]: same here

File: subgraphs/makerdao/src/utils/numbers.ts [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+    .times(exponent.minus(BIGDECIMAL_TWO))
+    .times(rate.times(rate).times(rate));
+  let fourthTerm = exponent
+    .div(BIGDECIMAL_TWELVE)
+    .times(exponent.minus(BIGDECIMAL_THREE))
+    .times(rate.times(rate).times(rate).times(rate));
+  return firstTerm.plus(secondTerm).plus(thirdTerm).plus(fourthTerm);
+}
+
+export function round(numberToRound: BigDecimal): BigDecimal {
```
Comment (this-username-is-taken) [inline subgraphs/makerdao/src/utils/numbers.ts:39]: and here

File: subgraphs/makerdao/src/utils/numbers.ts (hunk 1 above)
Comment (tnkrxyz) [reply subgraphs/makerdao/src/utils/numbers.ts:15]: Will do after I get the subgraph to work correctly. Some of the functions I borrowed from @[tannishmango](https://github.com/tannishmango)'s implement

Comment (tnkrxyz) [reply subgraphs/makerdao/src/utils/numbers.ts:15]: Looks like this function is already in there `_reference/src/common/utils/numbers.ts`: `bigIntToBigDecimal`

**Comment by tnkrxyz** [comment]
A temporary deployment of the current implement is available at: https://api.studio.thegraph.com/query/29379/makerdao-ethereum-v2/v0.0.2

A few things to note:
- Since the lending schema expects supply side revenue from a market, a 'market' is created for the DAI saving contract POT, even though it is a real market.
- About 400 liquidations have negative profitUSD, one as high as -$17,381.73 (liquidate.id=0x8f06a94a522dff90a445eadc7569919abab4caee936e011aed1517acca2dc404-245). After double, triple check, as far as I can see the numbers are correct, but they seem odd. For example, for 0x8f06a94a522dff90a445eadc7569919abab4caee936e011aed1517acca2dc404-245, when it was flag for liquidation, 89.61790614077303 ETH was seized for a loan of $50,000. When the auction started, the bids went to 85.450979507487898432 ETH, 55.741671655855377187 ETH, and finally 42.361384320354736031 ETH after 25 rounds of bidding. With the price of ETH at $754.01, the collateral worth $31,940.91 and a net loss of -$17,381 to the liquidator.

**Comment by this-username-is-taken** [comment]
Really good fixes. Thanks for being so meticulous with this one!

**Pull request merged**